src/python/WMCore/MicroService/Tools/PycurlRucio.py

```diff
@@ -188,16 +199,24 @@ def listReplicationRules(containers, rucioAccount, grouping,
                         logging.warning(msg)
                         continue
 
-                    timeDiff = item['stuck_at'] - item['created_at']
-                    if int(timeDiff.days) > STUCK_LIMIT:
+                    # then calculate for how long it's been stuck
+                    utcTimeNow = int(datetime.datetime.utcnow().strftime('%s'))
+                    if item['stuck_at']:
+                        stuckAt = stringDateToEpoch(item['stuck_at'])
```

Comment:
I see the benefits from the attempt to calculate the time difference in seconds, but what I am concerned here is that the method `stringDateToEpoch()` does not have any protection from receiving `None` as an argument [1]. And it seems to me this was  the original cause of the problem. So I would say here we will still get the Exception. 

BTW Do we know the conditions under which the `item['stuck_at']` value becomes `None`?

[1]
```
In [4]: stringDateToEpoch(None)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-ced2c9ac0b87> in <module>()
----> 1 stringDateToEpoch(None)

<ipython-input-1-a4fbc704f9e9> in stringDateToEpoch(strDate)
      7     :return: the equivalent EPOCH time (integer)
      8     """
----> 9     timestamp = datetime.datetime.strptime(strDate, "%a, %d %b %Y %H:%M:%S %Z")
     10     return int(timestamp.strftime('%s'))

TypeError: strptime() argument 1 must be string, not None
```

Reply:
That's what the `if` statement at L204 is supposed to do. 
Given that `created_at` seems to be a property of the rule creation process, I believe that one would always have a non-None value, hence that parameter is not really protected from a None value. Different than `stuck_at`, which is likely initialized with None value.

Reply:
I see, that would do the job. It seems we are safe to continue here, but I am curious and we should forward an inquiry to Rucio folks, asking how those values are initialized and changed throughout the transfer life time.

Reply:
Pushed the wrong button, for the record: https://github.com/rucio/rucio/issues/4592